reactor-core/src/main/java21/reactor/core/scheduler/LoomBoundedElasticScheduler.java

```diff
@@ -0,0 +1,1394 @@
+			int stateQueueSize = state.estimateQueueSize();
+			if (stateQueueSize >= 0) {
+				totalTaskCapacity -= stateQueueSize;
+			}
+			else {
+				return -1;
+			}
+		}
+		return totalTaskCapacity;
+	}
+
+	@Override
+	public Object scanUnsafe(Attr key) {
+		if (key == Attr.TERMINATED || key == Attr.CANCELLED) return isDisposed();
+		if (key == Attr.BUFFERED) return estimateSize();
+		if (key == Attr.CAPACITY) return maxThreads;
+		if (key == Attr.NAME) return this.toString();
+
+		return null;
+	}
+
+	@Override
+	public Stream<? extends Scannable> inners() {
+		BoundedServices services = state.currentResource;
+		return Stream.concat(Stream.of(services.busyStates.array), services.idleQueue.stream())
+		             .filter(obj -> obj != null && obj != CREATING);
+	}
+
+	@Override
+	public Worker createWorker() {
+		return state.currentResource.pick();
+	}
+
+	static final class BoundedServices extends AtomicInteger {
+
+		static final class BusyStates {
+			final BoundedState[] array;
+			final boolean shutdown;
+
+			public BusyStates(BoundedState[] array, boolean shutdown) {
+				this.array = array;
+				this.shutdown = shutdown;
+			}
+		}
+
+		/**
+		 * The {@link ZoneId} used for clocks. Since the {@link Clock} is only used to ensure
+		 * TTL cleanup is executed every N seconds, the zone doesn't really matter, hence UTC.
+		 *
+		 * @implNote Note that {@link ZoneId#systemDefault()} isn't used since it triggers disk read,
+		 * contrary to {@link ZoneId#of(String)}.
+		 */
+		static final ZoneId ZONE_UTC = ZoneId.of("UTC");
+
+		static final BusyStates ALL_IDLE = new BusyStates(new BoundedState[0], false);
+		static final BusyStates ALL_SHUTDOWN = new BusyStates(new BoundedState[0], true);
+		static final ScheduledExecutorService EVICTOR_SHUTDOWN;
+
+		static final BoundedServices SHUTDOWN;
+		static final BoundedServices SHUTTING_DOWN;
+		static final BoundedState    CREATING;
+
+		static {
+			EVICTOR_SHUTDOWN = Executors.newSingleThreadScheduledExecutor();
+			EVICTOR_SHUTDOWN.shutdownNow();
+
+			SHUTDOWN = new BoundedServices();
+			SHUTTING_DOWN = new BoundedServices();
+			SHUTDOWN.dispose();
+			SHUTTING_DOWN.dispose();
+
+			CREATING = new BoundedState(SHUTDOWN) {
+				@Override
+				public String toString() {
+					return "CREATING BoundedState";
+				}
+			};
+			CREATING.markCount = -1; //always -1, ensures tryPick never returns true
+			CREATING.idleSinceTimestamp = -1; //consider evicted
+		}
+
+		static final AtomicLong EVICTOR_COUNTER = new AtomicLong();
+		static final ThreadFactory EVICTOR_FACTORY = r -> {
+			Thread t = new Thread(r, Schedulers.BOUNDED_ELASTIC + "-evictor-" + EVICTOR_COUNTER.incrementAndGet());
+			t.setDaemon(true);
+			return t;
+		};
+
+
+		final LoomBoundedElasticScheduler parent;
+		//duplicated Clock field from parent so that SHUTDOWN can be instantiated and partially used
+		final Clock                       clock;
+		final ScheduledExecutorService    evictor;
```

Comment:
We probably don't need the evictor, as we no longer have a heavy `Thread` to release, we could just rely on garbage collection and re-create a new `BoundedState` every time it is required.

Reply:
fixed